### PR TITLE
Fix/716 tables tab

### PR DESCRIPTION
**Pull request by HarryMytilinaios**
The below commit adds a `Tables Tab` in the LeftPanel Tabs Summary. The `Tables Tab` follows the implementation of the `DataDrawer` component. Some refactoring have been made in order to facilitate the changes. Some of those are:

- Deleted `DataDrawer` component as it wasn't needed anymore.
- Rearranged  a little bit all four tab panels components (ChartsPanel, AnalysisPanel, LayersPanel, TablesPanel).
- The layers and tables are being filtered and feeded inside each individual component directly from the parent, which filters the `menuList` accordingly.
- `utils` file which was in the `LayersPanel` component has been moved to `LeftPanelTabs` component because the `menuList`, which was exported is now being processed in the `LeftPanelTabs` component.
- Moved `Chart` component to `Common` folder.

Practically when a country has tables the `Tables Tab` with `Tables Panel` will be added in the `LeftPanelMenu` and will display all the available tables in the dropdown menu.

This touches quite a few components so let me know if anything isn't clear.  Cc @ericboucher @wadhwamatic .

Below I attach some screenshots.

Screenshot/video of feature:
Screenshot from 2023-04-26 14-47-01
Screenshot from 2023-04-26 14-47-08
Screenshot from 2023-04-26 14-47-14
Screenshot from 2023-04-26 14-47-30
Screenshot from 2023-04-26 14-47-38

### CODE REVIEW

**Comment by HarryMytilinaios** [comment]
I am adding also a screenshot below with all four tabs in place. If we have a specific country that has also layers,charts,analysis and tables.

Screenshot from 2023-04-26 15-20-24
Screenshot from 2023-04-26 15-22-10
Screenshot from 2023-04-26 15-22-13
Screenshot from 2023-04-26 15-22-18

**Comment by wadhwamatic** [comment]
great work @HarryMytilinaios! 

I temporarily added a table config for Cambodia and a chart config for Mongolia to see how charts / tables work together. It's working for Mongolia (national level only because the admin codes aren't there for admin1 and admin2). 

I couldn't get tables to show up for Cambodia. I might need a reminder on how this is setup. I thought it just required adding the rawTables reference in src/config/country/index.ts, creating tables.json, and adding tables to prism.json. What else am I missing? cc @ericboucher

**Comment by HarryMytilinaios** [comment]
@wadhwamatic I think this is now fixed. The problem was that the menuList that we are used to populate both layers and tables was expeting a `categories` key inside the `prism.json` which will include both layers and tables. There was a typo in the `prism.json` of cambodia and the tables instead of being a child of `categories` was at the same level as `categories` as siblings.

File: frontend/public/data/cambodia/tables/2021-02-06/NSO_Camel_Price_Monthly_2021.csv [hunk 1]
```diff
@@ -0,0 +1,157 @@
+﻿Category_MN,Category_EN,Admin_Code,Month1,Month2,Month3,Month4,Month5,Month6,Month7,Month8,Month9,Month10,Month11,Month12
```
Comment (ericboucher) [inline frontend/public/data/cambodia/tables/2021-02-06/NSO_Camel_Price_Monthly_2021.csv:1]: @wadhwamatic please remove the extra data once you've confirmed it is working as you would expect :)

File: frontend/src/components/MapView/LeftPanel/LeftPanelTabs/index.tsx [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+    }, [activeLayers.length, panelSize, t, tabValue]);
+
+    const a11yProps = useCallback((index: any) => {
+      return {
+        id: `full-width-tab-${index}`,
+        'aria-controls': `full-width-tabpanel-${index}`,
+      };
+    }, []);
+
+    const renderedTabWidth = useMemo(() => {
```
Comment (ericboucher) [inline frontend/src/components/MapView/LeftPanel/LeftPanelTabs/index.tsx:145]: You could simply do something like:

`const nTabs = 2 + sum(reChartLayersAvailable, areTablesAvailablem)`
return `'calc(100% / nTabs)`

File: frontend/src/config/mongolia/prism.json [hunk 3]
```diff
@@ -1,5 +1,6 @@
 {
   "country": "Mongolia",
+  "countryAdmin0Id": 44,
```
Comment (ericboucher) [inline frontend/src/config/mongolia/prism.json:3]: let's add the real number here,  `167`

File: frontend/src/config/mongolia/layers.json [hunk 4]
```diff
@@ -377,6 +377,35 @@
       "pixelResolution": 64,
       "offset": -200
     },
+    "chart_data": {
```
Comment (ericboucher) [inline frontend/src/config/mongolia/layers.json:380]: delete before merging

File: frontend/src/config/cambodia/tables.json [hunk 5]
```diff
@@ -0,0 +1,93 @@
+{
```
Comment (ericboucher) [inline frontend/src/config/cambodia/tables.json:1]: revert before merging

File: frontend/src/config/cambodia/prism.json [hunk 6]
```diff
@@ -203,6 +203,18 @@
     },
     "field_reports": {
       "field_reports": ["flood_report", "drought_report", "kh_incident_report"]
+    },
+    "tables": {
```
Comment (ericboucher) [inline frontend/src/config/cambodia/prism.json:207]: revert before merging

**Review by ericboucher — APPROVED** [review]

File: frontend/src/components/MapView/LeftPanel/LeftPanelTabs/index.tsx (hunk 2 above)
Comment (HarryMytilinaios) [reply frontend/src/components/MapView/LeftPanel/LeftPanelTabs/index.tsx:145]: Yes you are right on this changing it.

File: frontend/src/config/mongolia/prism.json (hunk 3 above)
Comment (wadhwamatic) [reply frontend/src/config/mongolia/prism.json]: removed this as we're not including charts in mongolia

File: frontend/public/data/cambodia/tables/2021-02-06/NSO_Camel_Price_Monthly_2021.csv (hunk 1 above)
Comment (wadhwamatic) [reply frontend/public/data/cambodia/tables/2021-02-06/NSO_Camel_Price_Monthly_2021.csv]: done

File: frontend/src/config/cambodia/tables.json (hunk 5 above)
Comment (wadhwamatic) [reply frontend/src/config/cambodia/tables.json]: done

File: frontend/src/config/cambodia/prism.json (hunk 6 above)
Comment (wadhwamatic) [reply frontend/src/config/cambodia/prism.json]: done

**Comment by wadhwamatic** [comment]
Thanks for catching the issue and resolving it @HarryMytilinaios. Everything looks good from my side now. I removed the test configs for Cambodia and Mongolia. This is GTG from my side

**Review by wadhwamatic — APPROVED** [review]

**Pull request merged**